src/love.cpp

```diff
@@ -299,6 +299,35 @@ static DoneAction runlove(int argc, char **argv, int &retval, love::Variant &res
 	return done;
 }
 
+#ifdef LOVE_WINDOWS
+static INT_PTR CALLBACK EmptyDlgProc(HWND, UINT, WPARAM, LPARAM)
+{
+    return 0;
+}
+
+// Windows stores an initial window state passed in by the shell (or another process) if STARTF_USESHOWWINDOW is set.
+// This state is applied to the first window that receives a 'ShowWindow' call.
+// To ensure our window is not influenced by this, we create a dummy window to catch this initial state.
+// Ideally we would use SDL for this, but it skips calls to 'ShowWindow' if the window is hidden...
+static void love_windows_ignore_initial_wShowWindow()
```

Comment:
Usually we don't do this sort of platform-specific workaround and leave it up to SDL instead - this sort of thing can be difficult for us to maintain and might have unexpected side effects.

I think what I'd expect from SDL is:
* If the first visible window is resizable and not fullscreen, the "start maximized" size is applied (with the correct size).
* Otherwise it's ignored.

If SDL3 still has behaviour similar to #1714, maybe a bug report can be filed there? And any code changes to fix or work around the issue could be applied to SDL's codebase rather than love's.

Reply:
The main window in SDL3 does not seem affected, the compatibility messagebox does.

I have filed an issue: https://github.com/libsdl-org/SDL/issues/9434

Reply:
Thanks for investigating!

Reply:
SDL has added the [3.2.0](https://github.com/libsdl-org/SDL/milestone/10) milestone on that ticket.